frontend/app/routes/_public+/apply+/$id+/personal-information.tsx

```diff
@@ -31,50 +36,234 @@ export async function loader({ request, params }: LoaderFunctionArgs) {
+      homePostalCode: z.string().min(1, { message: 'empty-field' }).trim(),
+    })
+    .superRefine((val, ctx) => {
+      if (!isValidPostalCode(val.mailingCountry, val.mailingPostalCode)) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: val.mailingCountry === 'CAN' ? 'invalid-postal-code' : 'invalid-zip-code',
+          path: ['mailingPostalCode'],
+        });
+      }
+      if (!isValidPostalCode(val.homeCountry, val.homePostalCode)) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: val.homeCountry === 'CAN' ? 'invalid-postal-code' : 'invalid-zip-code',
+          path: ['homePostalCode'],
+        });
+      }
+    });
+
+  //schema for copy mailing address to home address
+  const mailingAddressOnlyFormSchema = z
+    .object({
+      phoneNumber: z
+        .string()
+        .refine((val) => !val || (val && isValidPhoneNumber(val, 'CA')), { message: 'invalid-phone' })
+        .optional(),
+      phoneNumberAlt: z
+        .string()
+        .refine((val) => !val || (val && isValidPhoneNumber(val, 'CA')), { message: 'invalid-phone' })
+        .optional(),
+      mailingAddress: z
+        .string()
+        .min(1, { message: 'empty-field' })
+        .transform((val) => val.trim()),
+      mailingApartment: z.string().trim().optional(),
+      mailingCountry: z
+        .string()
+        .min(1, { message: 'empty-field' })
+        .transform((val) => val.trim()),
+      mailingProvince: z.string().min(1, { message: 'empty-field' }).optional(),
+      mailingCity: z
+        .string()
+        .min(1, { message: 'empty-field' })
+        .transform((val) => val.trim()),
+      mailingPostalCode: z.string().min(1, { message: 'empty-field' }).trim(),
+      copyMailingAddress: z.string().optional(),
+      homeAddress: z.string().optional(),
+      homeApartment: z.string().optional(),
+      homeCountry: z.string().optional(),
+      homeProvince: z.string().optional(),
+      homeCity: z.string().optional(),
+      homePostalCode: z.string().optional(),
+    })
+    .superRefine((val, ctx) => {
+      if (!isValidPostalCode(val.mailingCountry, val.mailingPostalCode)) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: val.mailingCountry === 'CAN' ? 'invalid-postal-code' : 'invalid-zip-code',
+          path: ['mailingPostalCode'],
+        });
+      }
+    });
+
   const formData = Object.fromEntries(await request.formData());
-  const parsedDataResult = applyFlow.personalInfoStateSchema.safeParse(formData);
+  const { copyMailingAddress } = formData as Partial<z.infer<typeof copyMailingAddressSchema>>;
+
+  let parsedDataResult;
+  //only parse mailing address since home address will be copied
+  if (copyMailingAddress === 'on') {
+    parsedDataResult = mailingAddressOnlyFormSchema.safeParse(formData);
+  } else {
+    parsedDataResult = personalInformationFormSchema.safeParse(formData);
+  }
```

Comment:
**Suggestion:** It would be more maintainable to have a single validation schema that takes care of both scenarios. 

Here's my attempt: https://github.com/DTS-STN/canadian-dental-care-plan/commit/64ccb5e64987fb102d828d6005f3fa748bbc6260